### PR TITLE
DML EP Bitwise operators opset 18

### PR DESCRIPTION
**Pull request by linnealovespie**
### Description

Add dml registration for bitwise and, or, xor and not added in opset 18.

### CODE REVIEW

File: onnxruntime/test/providers/cpu/math/element_wise_ops_test.cc [hunk 1]
```diff
@@ -2651,87 +2651,87 @@ void TrigFloat16Test(OpTester& test, std::initializer_list<float> input) {
...
-TEST(MathOpTest, Acosh) {
-  OpTester test("Acosh", 9);
-  TrigFloatTest<::acoshf>(test, {1.0f, 1.1f, 3.0f, 10.0f, 100.0f});
-}
-
-TEST(MathOpTest, Atanh) {
-  OpTester test("Atanh", 9);
-  TrigFloatTest<::atanhf>(test, {-1.0f, -0.5f, 0.0f, 0.5f, 1.0f});
-}
+// TEST(MathOpTest, SinFloat) {
```
Comment (fdwr) [inline onnxruntime/test/providers/cpu/math/element_wise_ops_test.cc:2654]: 🤨 These changes are unrelated to BitwiseAnd.

File: onnxruntime/test/providers/cpu/math/element_wise_ops_test.cc [hunk 2]
```diff
@@ -3076,5 +3076,267 @@ TEST(BitShiftOpTest, BroadcastXRight_Uint8) {
   test.Run();
 }
 
+TEST(MathOpTest, BitwiseAnd) {
```
Comment (fdwr) [inline onnxruntime/test/providers/cpu/math/element_wise_ops_test.cc:3079]: Hmm, it saddens me to see c++ code written to test these than *only* exist within ORT, rather than data-driven .onnx files which would benefit *everybody* implementing the standard (e.g. https://github.com/onnx/onnx/tree/main/onnx/backend/test/data/node/test_bitwise_not_2d) or JSON files which aren't C++ specific (like our WindowsAI conformance test suite, which I pointed the task to Numfor about). Of course, I'm still will gladly accept additional test collateral, but it's current form doesn't have the biggest impact/benefit - something to think about...

File: onnxruntime/test/providers/cpu/math/element_wise_ops_test.cc [hunk 3]
```diff
@@ -3076,5 +3076,267 @@ TEST(BitShiftOpTest, BroadcastXRight_Uint8) {
...
+
+TEST(MathOpTest, BitwiseXor_multidiretional_broadcastAB) {
+  OpTester test("BitwiseXor", 18);
+  test.AddInput<int32_t>("A", {4, 1}, {10, 11, 12, 13});
+  test.AddInput<int32_t>("B", {2}, {15, 7});
+  test.AddOutput<int32_t>("C", {4, 2}, {5, 13, 4, 12, 3, 11, 2, 10});
+  test.Run();
+}
+
+TEST(MathOpTest, BitwiseXor_multidiretional_broadcastBA) {
```
Comment (fdwr) [inline onnxruntime/test/providers/cpu/math/element_wise_ops_test.cc:3302]: `multidirectional`

**Review by fdwr — COMMENTED** [review]
The `MathOpTest, SinFloat` part looks fishy and unrelated, but otherwise good. Thanks for adding.

**Review by fdwr — APPROVED** [review]

**Review by smk2007 — APPROVED** [review]

**Review by fdwr — APPROVED** [review]
👍 Good, the MathOpTest tests are restored.

**Review by fdwr — DISMISSED** [review]
Merge issue with Sheil's? LGTM.

**Review by smk2007 — APPROVED** [review]
